src/Context/WatchLater.context.js

```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
+import { AlertToast, SuccessToast } from "Components";
+import { createContext, useContext } from "react";
+import { useReducerContext } from "./Reducer.context";
+
+const WatchLaterContext = createContext();
+
+const WatchLaterProvider = ({ children }) => {
+  const encodedToken = localStorage.getItem("StormPlayToken");
+  const { dispatch } = useReducerContext();
+
+  const addToWatchLater = async (video) => {
+    try {
+      const response = await axios.post(
+        "/api/user/watchlater",
+        { video },
+        { headers: { authorization: encodedToken } }
+      );
+      if (response.status === 201) {
+        SuccessToast("Added To Watch Later");
+        dispatch({ type: "WATCH_LATER", payload: response.data.watchlater });
+      }
+    } catch (error) {
+      console.log(error);
```

Comment:
Suggestion: You can dispatch a failure toast

Reply:
thanks for pointing it out will make sure to dispatch a failure toast